packages/world/src/WorldProxy.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { StoreCore } from "@latticexyz/store/src/StoreCore.sol";
+import { STORE_VERSION } from "@latticexyz/store/src/version.sol";
+import { IStoreEvents } from "@latticexyz/store/src/IStoreEvents.sol";
+import { WORLD_VERSION } from "./version.sol";
+import { IWorldEvents } from "./IWorldEvents.sol";
+import { AccessControl } from "./AccessControl.sol";
+import { ROOT_NAMESPACE_ID } from "./constants.sol";
+import { Proxy } from "./Proxy.sol";
+import { IERC1967 } from "./IERC1967.sol";
+import { StorageSlot } from "./StorageSlot.sol";
+
+/**
+ * @dev Storage slot with the address of the current implementation.
+ * This is the keccak-256 hash of "eip1967.proxy.implementation" subtracted by 1.
+ */
+bytes32 constant IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+
+/**
+ * @dev Stores a new address in the EIP1967 implementation slot.
+ */
+function _setImplementation(address newImplementation) {
+  StorageSlot.getAddressSlot(IMPLEMENTATION_SLOT).value = newImplementation;
+
+  emit IERC1967.Upgraded(newImplementation);
+}
```

Comment:
why is this a free function rather than an internal function on WorldProxy?

Reply:
I generally prefer free functions when possible! but open to moving it

Reply:
why do you prefer free functions? have opted for internal functions in MUD so far, don't have a strong preference, but unless there is a reason for free functions i'd say let's move it to an internal one for consistency with the rest of the codebase

Reply:
Agreed, done